test/integration/job/job_test.go

```diff
@@ -1762,59 +1763,97 @@ func TestJobPodReplacementPolicy(t *testing.T) {
 			defer cancel()
 			resetMetrics()
 
-			jobObj, err := createJobWithDefaults(ctx, clientSet, ns.Name, &batchv1.Job{
-				Spec: *tc.jobSpec,
+			runJobAndVerifyStatusPodReplacement(ctx, podCount, tc.jobSpec, t, clientSet, "test-job", ns, &podsByStatus{
+				Terminating: tc.wantTerminating,
+				Failed:      tc.wantFailed,
+				Active:      tc.wantActive,
+				Ready:       pointer.Int32(0),
 			})
-			if err != nil {
-				t.Fatalf("Failed to create Job: %v", err)
-			}
-			jobClient := clientSet.BatchV1().Jobs(jobObj.Namespace)
+		})
+	}
+}
 
-			// Wait for pods to start up.
-			err = wait.PollImmediate(5*time.Millisecond, wait.ForeverTestTimeout, func() (done bool, err error) {
-				job, err := jobClient.Get(ctx, jobObj.Name, metav1.GetOptions{})
-				if err != nil {
-					return false, err
-				}
-				if job.Status.Active == int32(podCount) {
-					return true, nil
-				}
-				return false, nil
+func TestJobPodReplacementPolicyFeatureToggling(t *testing.T) {
+	const podCount int32 = 2
+	nonIndexedCompletion := batchv1.NonIndexedCompletion
+	var podReplacementPolicy = func(obj batchv1.PodReplacementPolicy) *batchv1.PodReplacementPolicy {
+		return &obj
+	}
+	cases := map[string]struct {
+		wantTerminating *int32
+		wantActive      int
+		wantFailed      int
+		jobSpec         *batchv1.JobSpec
+	}{
+		"podReplacementPolicy failed; restart and disable feature; renable feature": {
+			jobSpec: &batchv1.JobSpec{
+				Parallelism:          ptr.To(podCount),
+				Completions:          ptr.To(podCount),
+				CompletionMode:       &nonIndexedCompletion,
+				PodReplacementPolicy: podReplacementPolicy(batchv1.Failed),
+			},
+			wantTerminating: ptr.To(podCount),
+		},
+		"podReplacementPolicy terminatingOrFailed; restart and disable feature; renable feature": {
+			jobSpec: &batchv1.JobSpec{
+				Parallelism:          ptr.To(podCount),
+				Completions:          ptr.To(podCount),
+				CompletionMode:       &nonIndexedCompletion,
+				PodReplacementPolicy: podReplacementPolicy(batchv1.TerminatingOrFailed),
+			},
+			wantTerminating: ptr.To(podCount),
+			wantActive:      int(podCount),
+			wantFailed:      int(podCount),
+		},
+	}
+	for name, tc := range cases {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.JobPodReplacementPolicy, true)()
+			closeFn, restConfig, clientSet, ns := setup(t, "pod-replacement-policy")
+			defer closeFn()
+			ctx, cancel := startJobControllerAndWaitForCaches(restConfig)
+			defer func() {
+				cancel()
+			}()
+			resetMetrics()
+
+			// Create a job with pod replacement policy set.
+			// Set Finalizers and delete pods
+			// Verify that terminating field is set and depending on Failed/TerminatingOrFailed check if pod is considered failed.
+			// We ran into problems doing this in the same test so we are going with creating a new job after restart.
```

Comment:
But the point of the test is to verify that an existing job can recover.
What problem did you run into?

Reply:
I see :)